hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -1060,6 +1061,30 @@ func (r *HostedClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		}
 	}
 
+	// Reconcile External CCM once kubeconfig is available
+	if hcp.Status.KubeConfig != nil {
+		kubeConfig := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: hcp.Namespace,
+				Name:      hcp.Status.KubeConfig.Name,
+			},
+		}
+		err := r.Client.Get(ctx, client.ObjectKeyFromObject(kubeConfig), kubeConfig)
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to get controlplane kubeconfig secret %q: %w", client.ObjectKeyFromObject(kubeConfig), err)
+		}
+
+		ccmImage, err := r.getPayloadImage(ctx, hcluster, ImageStreamPowerVSCloudControlManagerImage)
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to reconcile hostedcluster ccm while getting image id: %w", err)
+		}
```

Comment:
not comfortable with this one, it will be good if we can offload this code to the platform's `ReconcileExternalCCM`

Reply:
`getPayloadImage` is belongs to `r *HostedClusterReconciler`, we need to pass that as an arg to `ReconcileExternalCCM` to access `getPayloadImage`.  can I add switch block here to get the ccm image based on platform?

Reply:
Not a big fan of having switch case here, its looks odd.. wondering if we can send `getPayloadImage` function as param to `ReconcileExternalCCM` and each individual platform will send the relevant image name to find an image.

@enxebre wdys?

Reply:
Moved the CCM deployment as part of CPO controller. ptal.